### DISCUSSION TITLE
Version - where is it stored and what scheme to use?

### DISCUSSION THREAD

**Discussion by leo-amelyan** [grate-devs/grate#420] [Q&A] [answered]
Using the following to run migrations:

grate.exe --files .\scripts --databasetype postgresql --connectionstring "...'" --version 0.1.0 --silent --outputPath .\migration-output

Questions:

Where does the version get stored in the database? 

What versioning scheme is recommended?  I see that the version has four digits 0.0.0.0  and I am used to SEM versioning of just three major, minor, patch.

**Comment by hoangthanh28** [comment, marked as answer]
From what I can see, the versioning schema follows [assembly version](https://learn.microsoft.com/en-us/dotnet/standard/assembly/versioning) which is defined by: `<major version>.<minor version>.<build number>.<revision>`.
The main mechanism of grate is getting from the runtime assembly (by default) or user input via --version parameter (which you can place whatever as string).
I also catch some demand to follow the SEM version. I would suggest that you use the same version with your application. If your application is versioning follow SEM version -> use --version SEM_version in grate as well.

**Reply by leo-amelyan** [reply]
Thank you. I will use the SEM versioning since my other projects also use it. 

Also, I found how Grate stores the versioning. In the database of course! Creates it's own tables to track things.

![image](https://github.com/erikbra/grate/assets/1872686/cce4ee85-d300-48df-b996-0ba160b9bab8)
